src/data/emit/type.jl

```diff
@@ -9,6 +9,7 @@
         name=Symbol("typeof($(info.def.head.name))"),
         fields=[JLField(; name=:data, type=:(Union{$(union_params...)}))],
         typevars=info.whereparams,
+	supertype=info.def.head.supertype,
```

Comment:
so basically you just need to make it available and perhaps double check if `supertype` is a `nothing` or not.

```suggestion
	supertype=isnothing(info.def.head.supertype) ? nothing : Base.eval(info.mod, info.def.head.supertype),
```

Reply:
I don't think it is that simple. Evaluation of the type has to be delayed until the emission of the code. `info.def.head` points to `..`, which is not available within `Moshi.Data`.

Doing it like the suggestion errors with:

"LoadError: UndefVarError: `..` not defined in `Moshi.Data`"

Reply:
My guess is that something resembling [this](https://github.com/Roger-luo/Moshi.jl/blob/a68cee1647cf1374d5c1353ea8ca34b6627e49f8/src/data/emit/cons.jl#L42) or [this](https://github.com/Roger-luo/Moshi.jl/blob/a68cee1647cf1374d5c1353ea8ca34b6627e49f8/src/data/scan.jl#L35) is needed to `eval` the supertype in the right scope, and then pass it along the EmitInfo properly resolved. Is this so?

Reply:
yeah you are right, `guess_self_as_any`

```julia
supertype=isnothing(info.def.head.supertype) ? nothing : guess_self_as_any(info.def, info.def.head.supertype)
```